### PR TITLE
feature:  Protected B updates

### PR DESCRIPTION
**Pull request by timarney**
# Summary | Résumé

- Add protected B option to dropdown
- Re-add modal`Help on choosing an option`
- Added `Protected B`  message above radio options when `Protected B` is selected
- Updated publish screen to include classification 

See screen in comments below:

# Test instructions | Instructions pour tester la modification

- Test updating / adding a settings delivery
- Test publish checklist screen

https://yhahfutujmygvpjkrfyfwkjfya0ohoyt.lambda-url.ca-central-1.on.aws/

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [ ] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [ ] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [ ] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [ ] Have you modified the change log and updated any relevant documentation?
- [ ] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [ ] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

**Comment by timarney** [comment]
Re-added modal

Screenshot 2023-07-24 at 11 38 42 AM

Screenshot 2023-07-24 at 11 38 34 AM

File: public/static/locales/fr/form-builder.json [hunk 1]
```diff
@@ -399,7 +399,10 @@
   "publishingRemovesTestResponses": "La publication supprime les réponses test.",
   "publishingRemovesTestResponsesDescription": "Toutes les réponses antérieures seront supprimées du système.",
   "publishYourForm": "Publier votre formulaire",
-  "publishYourFormInstructions": "Vous devez franchir les étapes suivantes avant de publier votre formulaire. ",
+  "publishYourFormInstructions": {
```
Comment (timarney) [inline public/static/locales/fr/form-builder.json:402]: @anikbrazeau couple of strings here

File: public/static/locales/fr/form-builder.json [hunk 2]
```diff
@@ -452,6 +455,8 @@
   "settingsResponseDelivery": {
     "selectClassification": "Sélectionnez la classification des données des réponses de votre formulaire",
     "beforePublishMessage": "Vous pouvez modifier ces paramètres à tout moment avant de publier le formulaire.",
+    "protectedBMessage": "Pour modifier ces paramètres, sélectionnez la classification des réponses de votre formulaire ci-dessus.",
+    "settingsMessage": "[FR] To change these settings, select the data classification of your form responses.",
```
Comment (timarney) [inline public/static/locales/fr/form-builder.json:459]: @anikbrazeau string here :)

Comment (anikbrazeau) [inline public/static/locales/fr/form-builder.json:459]:
```suggestion
    "settingsMessage": "Pour modifier ces paramètres, sélectionnez la classification des données de vos réponses au formulaire.",
```

File: public/static/locales/en/form-builder.json [hunk 3]
```diff
@@ -451,6 +454,8 @@
   "settingsResponseDelivery": {
     "selectClassification": "Select the data classification of your form responses",
     "beforePublishMessage": "You can change these settings anytime before publishing.",
+    "protectedBMessage": "PROTECTED B class forms must be downloaded from GC Forms.",
```
Comment (anikbrazeau) [inline public/static/locales/en/form-builder.json:457]:
```suggestion
    "protectedBMessage": "PROTECTED B responses must be downloaded from GC Forms.",
```

**Review by craigzour — DISMISSED** [review]
Reviewed and tested! Other than that specific issue we have talked about (that is not new to this PR and that prevents a user from saving new form settings if the form was not already auto saved once) we should be good to include in the upcoming release :)

**Review by craigzour — APPROVED** [review]

**Pull request merged**